src/algebra/star/algebra.lean

```diff
@@ -32,6 +35,22 @@ class star_algebra (R : Type u) (A : Type v)
   star (r • a) = star r • star a :=
 star_algebra.star_smul r a
 
+instance {R₁ R₂ : Type*} [semiring R₁] [semiring R₂] (e : R₁ ≃+* R₂) :
+  ring_hom_inv_pair (↑e : R₁ →+* R₂) ↑e.symm :=
+⟨e.symm_to_ring_hom_comp_to_ring_hom, e.symm.symm_to_ring_hom_comp_to_ring_hom⟩
```

Comment:
@dupuisf, do we have this instance somewhere already?

Reply:
Why is this in `algebra/star/algebra`?  It's not using the star structure.  I think it should be defined much earlier.  But since the whole point of `ring_hom_inv_pair` is to allow for silent replacement of `e.symm` by something propeq but not defeq to it, please either give this very low priority or make it a def.

Reply:
Or, in fact, the idiomatic way to do it would be to forget this instance and instead define (in `algebra/star/basic`)
```
instance {R₁ R₂ : Type*} [semiring R₁] [semiring R₁] [semiring R₂] [semiring R₁] (e : R₁ ≃+* R₂) :
  ring_hom_inv_pair (↑star_ring_equiv : R₁ →+* R₂) ↑star_ring_equiv :=
```

Reply:
I tried this, but lean can't find the instance if I specialize it like that. See the latest version I pushed, which has the working and non-working code side-by-side.

Reply:
I'll look at it later, but in the meantime look at how it's done in our "test" file `src/linear_algebra/conjugate_linear.lean` in the branch `semilinear-surjective`.

Reply:
OK, I looked at it, the problem is that that Lean has trouble identifying `(Rᵒᵖ)ᵒᵖ` back with `R`.  Someday this should be fixed by introducing an `[is_op R S]` typeclass.  In the short term, the solution is to work over commutative rings:

```
def star_comm_ring_equiv {R : Type*} [comm_semiring R] [star_ring R] : R ≃+* R :=
{ to_fun := star,
  map_mul' := λ x y, by rw [star_mul, mul_comm],
  .. star_add_equiv }

instance star_ring_equiv.ring_hom_inv_pair {R : Type*} [comm_semiring R] [star_ring R] :
  ring_hom_inv_pair (↑star_comm_ring_equiv : R →+* R) ↑star_comm_ring_equiv :=
ring_hom_inv_pair.of_ring_equiv _
```

Reply:
I think having a separate version for commutative rings is anyway desirable, so we can use it for complex conjugation without having to deal with this whole op business.

Reply:
Note that `star_comm_ring_equiv` already exists, it's `star_ring_aut`.

Reply:
> OK, I looked at it, the problem is that that Lean has trouble identifying `(Rᵒᵖ)ᵒᵖ` back with `R`.

I don't understand where `(Rᵒᵖ)ᵒᵖ` appears in this example

Reply:
The problem is not `(Rᵒᵖ)ᵒᵖ`, but rather the fact that we need two instance rather than one: we also need `ring_hom_inv_pair ↑star_ring_equiv.symm (↑(star_ring_equiv : R ≃+* Rᵒᵖ)`

Reply:
Why do we need both?

Reply:
If we don't we run into trouble with `linear_equiv.symm` because the parameters are then reversed.